src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplication.groovy

```diff
@@ -17,33 +17,29 @@ package com.bmuschko.gradle.docker
 
 import com.bmuschko.gradle.docker.tasks.image.Dockerfile
 import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.Nested
 
 class DockerJavaApplication {
-    String baseImage = 'openjdk:jre-alpine'
-    private final CompositeExecInstruction compositeExecInstruction = new CompositeExecInstruction()
-
-    /**
-     * Deprecated as per https://docs.docker.com/engine/deprecated/#maintainer-in-dockerfile
-     * Will be removed in 4.x release.
-     */
-    @Deprecated
-    String maintainer = System.getProperty('user.name')
-
-    /**
-     * Temporary solution to be able to create image without
-     * <code>MAINTAINER</code> and preserve backward compatibility.
-     * Will be removed in 4.x release.
-     */
-    boolean skipMaintainer
-
-    @Deprecated
-    Integer port = 8080
-    Set<Integer> ports
-    String tag
-
-    Integer[] getPorts() {
-        return ports != null ? ports : [port]
+    final Property<String> baseImage
+    final Property<String> maintainer
```

Comment:
This says above that `maintainer` will be removed. Isn't this the perfect time to remove it?

Reply:
Agreed.